tests/pytests/unit/pillar/test_netbox.py

```diff
@@ -2318,6 +2318,60 @@ def test_when_we_retrieve_everything_successfully_then_return_dict(
         assert actual_result == expected_result
 
 
+def test_when_we_retrieve_everything_successfully_then_return_dict_with_custom_destination_(
+    default_kwargs,
+    device_results,
+    no_results,
+    device_interfaces_list,
+    device_ip_results,
+    site_results,
+    site_prefixes,
+    proxy_details,
+    pillar_results,
+    connected_devices_results,
+):
+
+    expected_result = pillar_results
+
+    default_kwargs["virtual_machines"] = False
+    default_kwargs["interfaces"] = True
+    default_kwargs["interface_ips"] = True
+    default_kwargs["site_details"] = True
+    default_kwargs["site_prefixes"] = True
+    default_kwargs["proxy_return"] = True
+    default_kwargs["connected_devices"] = True
+    default_kwargs["destination_pillar_key"] = "netbox:minion"
+
```

Comment:
I just copied the full test so this could be cleaned up for sure. I can add a minimal response fixture that is dedicated for this test if wanted.

Reply:
Scratch that, I have now copied the fixture data also. In theory we could deduplicate this a little now if you would like.